File: codes/Java/TemperatureConverter.java
import java.util.Scanner;
import java.util.regex.Pattern;

public class TemperatureConverter {
    public static void main(String[] args) {
        Scanner input = new Scanner(System.in);
        System.out.println("Enter the temperature in following format 12 C or 12 F : ");
        String source = input.nextLine();
        Pattern reg = Pattern.compile("([+\\-]?[0-9]+[.]?[0-9]*)([C|F])?");
        var matches = reg.matcher(source);
        if (!matches.find()) {
            System.out.println("Invalid input");
            return;
        }
        var temp = Double.parseDouble(matches.group(1));
        var unit = matches.group(2);
        if (unit == null) {
            System.out.println("Invalid input");
            return;
        }
        if (unit.equals("C")) {
            System.out.println(temp + " C is " + (temp * 9 / 5 + 32) + " F");
        } else if (unit.equals("F")) {
            System.out.println(temp + " F is " + (temp - 32) * 5 / 9 + " C");
        } else {
            System.out.println("Invalid input");
        }
    }
}

File: codes/Python/temperature_converter.py
from termcolor import colored
import re

print(colored("""

████████╗███████╗███╗░░░███╗██████╗░  ░█████╗░░█████╗░███╗░░██╗██╗░░░██╗███████╗██████╗░████████╗
╚══██╔══╝██╔════╝████╗░████║██╔══██╗  ██╔══██╗██╔══██╗████╗░██║██║░░░██║██╔════╝██╔══██╗╚══██╔══╝
░░░██║░░░█████╗░░██╔████╔██║██████╔╝  ██║░░╚═╝██║░░██║██╔██╗██║╚██╗░██╔╝█████╗░░██████╔╝░░░██║░░░
░░░██║░░░██╔══╝░░██║╚██╔╝██║██╔═══╝░  ██║░░██╗██║░░██║██║╚████║░╚████╔╝░██╔══╝░░██╔══██╗░░░██║░░░
░░░██║░░░███████╗██║░╚═╝░██║██║░░░░░  ╚█████╔╝╚█████╔╝██║░╚███║░░╚██╔╝░░███████╗██║░░██║░░░██║░░░
░░░╚═╝░░░╚══════╝╚═╝░░░░░╚═╝╚═╝░░░░░  ░╚════╝░░╚════╝░╚═╝░░╚══╝░░░╚═╝░░░╚══════╝╚═╝░░╚═╝░░░╚═╝░░░
""", "cyan"))

print(colored("Usage : <value> [F | C (default)]", "green"))
print(colored("<> required | [] optional (P.S) Dont literally type out the brackets!\n\n\n", "blue"))


def start():

    source = input(colored("Enter the temperature : ", "yellow"))
    defaultFormat = "C"

    matches = re.match(r'([+\-]?[0-9]+[.]?[0-9]*)([C|F])?',source.replace(" ", ""), re.M | re.I)
    
    if not matches:
        print(colored("Invalid format : " + source, "red"))
        return
    if matches.group(2):
        defaultFormat = matches.group(2)

    target = float(matches.group(1))

    if defaultFormat.lower() == "c":
        temp = (9/5 * target) + 32
        print(
            colored(f"\nTemperature in (F) from (C) {str(round(target,2))} = {str(round(temp,2))} F", "blue"))
    elif defaultFormat.lower() == "f":
        temp = (target - 32) * (5/9)
        print(
            colored(f"\nTemperature in (C) from (F) {str(round(target,2))} = {str(round(temp,2))} C", "blue"))

start()